architecture/2024-05-radius-on-dapr.md

```diff
@@ -0,0 +1,324 @@
+| Building Block API | Distributed systems primitives provided by Dapr. Each building block has multiple components that implement it. Examples include state store, pub/sub, workflows. [docs](https://docs.dapr.io/developing-applications/building-blocks/).                                                                                                                              |
+| Component          | An implementation of a specific building block. For example, Dapr provides a Redis component that implements the state store building block. Component is also used as the term for Dapr's configuration format.                                                                                                                                                      |
+| Pub/Sub            | A messaging building-block that implements commonly-used [pub/sub](https://aws.amazon.com/what-is/pub-sub-messaging/#:~:text=The%20publish%2Dsubscribe%20(pub%2Fsub)%20model%20enables%20event,placed%20in%20a%20shopping%20cart.) functionality. See [docs](https://docs.dapr.io/reference/components-reference/supported-pubsub/) for supported pub/sub components. |
+| State store        | A key-value-store building block. See [docs](https://docs.dapr.io/reference/components-reference/supported-state-stores/) for supported state stores.                                                                                                                                                                                                                 |
+| Workflows          | A [workflow-as-code](https://docs.dapr.io/developing-applications/building-blocks/workflow/workflow-overview/) programming model for reliable distributed applications.                                                                                                                                                                                               |
+
+
+## Objectives
+
+> **Issue Reference:** <!-- (If appropriate) Reference an existing issue that describes the feature or bug. -->
+
+### Goals
+
+- Adopt Dapr for our data storage and messaging needs. 
+  - Keep `rad init` and `rad install` simple by including Dapr and a default choice of components as part of our Helm chart.
+  - Keep the Helm chart flexible by documenting how to BYO components.
+  - Provide production-guidance for Radius based on Dapr's existing guidance.
+- Remove our existing messaging and storage implementations in favor of Dapr.
+
+### Non goals
+
+- (out of scope) User-facing behavior changes other than installation/dependencies.
+- (out of scope/follow-up) Adoption of Dapr workflows.
+  - While this is highly valuable for us, this 
```

Comment:
Was this sentence meant to be completed?

Reply:
Yes :). I'll take a todo